File: 0. Script/Abstracts/12/Other/2/Programming/Script/1/1_0/aClass_Programming_ScriptRoutable_12_2_1_0.cs
using BaseDI.Professional.Script.Programming.Poco_1;
using BaseDI.Professional.Script.Programming.Repository_1;
using BaseDI.Professional.Script.Programming_1;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BaseDI.Professional.Script.Programming.Abstract_1
{
    public abstract class aClass_Programming_ScriptRoutable_12_2_1_0 
    {
        public abstract Task<JObject> Action();

        public Func<SingleParmPoco_12_2_1_0, dynamic> CallBack { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<Task<JObject>> Repository { get; set; }

        public JObject StorylineDetails { get; set; }

        public JObject StorylineDetails_Cached { get; set; }

        public JObject StorylineDetails_Parameters { get; set; }

        public ExtraData_12_2_1_0 ExtraData { get; set; }

        public aClass_Programming_ScriptRoutable_12_2_1_0 DirectorOrExperienceRequestHandler { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<JObject> MasterLeader { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<JObject> MasterController { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<JObject> MasterTransporter { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<JObject> MasterConverter { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<JObject> MasterStorer { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<JObject> MasterDisturber { get; set; }

        public aClass_Programming_ScriptAction_12_2_1_0<JObject> MasterSensor { get; set; }

        public aClass_Programming_ScriptNicheMaster_12_2_1_0<JObject> NicheMaster { get; set; }

        public Dictionary<string, object> ClientOrServerInstance { get; set; }

        public string RequestID { get; set; }
}
}

File: 0. Script/Abstracts/12/Other/2/Programming/Script/1/1_0/aClass_Programming_ScriptRoutable_12_2_1_0.ts
import * as aClass_Programming_ScriptAction_12_2_1_0 from "../../../../../../../../Abstracts/12/Other/2/Programming/Script/1/1_0/aClass_Programming_ScriptAction_12_2_1_0";
import * as aClass_Programming_ScriptRoutable_12_2_1_0 from "../../../../../../../../Abstracts/12/Other/2/Programming/Script/1/1_0/aClass_Programming_ScriptRoutable_12_2_1_0";

import * as ExtraData_12_2_1_0 from "../../../../../../../../Parameters/12/Other/2/Programming/ExtraData Poco/1/1_0/ExtraData_12_2_1_0";

import * as IContract_Programming_BaseDI_12_2_1_0 from "../../../../../../../../Interfaces/12/Other/2/Programming/Contract/1/1_0/IContract_Programming_BaseDI_12_2_1_0";

export namespace BaseDI.Programming.Abstract_1
{
    export abstract class aClass_Programming_ScriptRoutable_12_2_1_0
    {
        //#region 1. Assign
        public CallBack: (...args: any[]) => any = null;

        public Repository: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<any>;

        public StorylineDetails: object = new Object();

        public StorylineDetails_Parameters: object = new Object();

        public ExtraData: any;

        public EntryPoint: any = new Object();

        public MasterLeader: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<object>;

        public MasterController: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<object>;

        public MasterTransporter: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<object>;

        public MasterConverter: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<object>;

        public MasterStorer: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<object>;

        public MasterDisturber: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<object>;

        public MasterSensor: aClass_Programming_ScriptAction_12_2_1_0.BaseDI.Programming.Abstract_1.aClass_Programming_ScriptAction_12_2_1_0<object>;

        public ClientOrServerInstance: any;

        public RequestID: string;
        //#endregion

        //#region 2. Ready
        constructor()
        {
            //#region 1. Assign
            //#endregion

            //#region 2. Action
            //#endregion

            //#region 3. Observe
            //#endregion
        }
        //#endregion

        //#region 3. Set
        //#endregion

        //#region 4. Action
        public abstract async Action(): Promise<object>;
        //#endregion
    }
}